src/ngResource/resource.js

```diff
@@ -431,16 +431,20 @@ angular.module('ngResource', ['ng']).
             }
             if (!(new RegExp("^\\d+$").test(param)) && param &&
               (new RegExp("(^|[^\\\\]):" + param + "(\\W|$)").test(url))) {
-              urlParams[param] = true;
+              urlParams[param] = { isQueryParam: (new RegExp("\\?(.*)\\:" + param)).test(url) };
```

Comment:
I've updated the code accordingly and split my commit messages to better comply with contribution guidelines.